storage/src/test/java/org/apache/kafka/tiered/storage/specs/ConsumableSpec.java

```diff
@@ -16,33 +16,7 @@
  */
 package org.apache.kafka.tiered.storage.specs;
 
-import java.util.Objects;
-
-public final class ConsumableSpec {
-
-    private final Long fetchOffset;
-    private final Integer expectedTotalCount;
-    private final Integer expectedFromSecondTierCount;
-
-    public ConsumableSpec(Long fetchOffset,
-                          Integer expectedTotalCount,
-                          Integer expectedFromSecondTierCount) {
-        this.fetchOffset = fetchOffset;
-        this.expectedTotalCount = expectedTotalCount;
-        this.expectedFromSecondTierCount = expectedFromSecondTierCount;
-    }
-
-    public Long getFetchOffset() {
-        return fetchOffset;
-    }
-
-    public Integer getExpectedTotalCount() {
-        return expectedTotalCount;
-    }
-
-    public Integer getExpectedFromSecondTierCount() {
-        return expectedFromSecondTierCount;
-    }
+public record ConsumableSpec(Long fetchOffset, Integer expectedTotalCount, Integer expectedFromSecondTierCount) {
```

Comment:
I checked the usage of this class, and it appears using primitive types should be fine.

Reply:
Yep, I switched to primitive types in a bunch of classes